### PR TITLE
TYP: some types for pandas/_config/config.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
broken off #28339

### CODE REVIEW

File: pandas/_config/config.py [hunk 1]
```diff
@@ -455,7 +455,9 @@ def register_option(key, defval, doc="", validator=None, cb=None):
     path = key.split(".")
 
     for k in path:
-        if not bool(re.match("^" + tokenize.Name + "$", k)):
+        # NOTE: tokenize.Name is not a public constant
+        # error: Module has no attribute "Name"  [attr-defined]
+        if not bool(re.match("^" + tokenize.Name + "$", k)):  # type: ignore
```
Comment (jbrockmendel) [inline pandas/_config/config.py]: i think the `bool` is unnecessary

File: pandas/_config/config.py [hunk 2]
```diff
@@ -455,7 +455,9 @@ def register_option(key, defval, doc="", validator=None, cb=None):
...
             raise ValueError("{k} is not a valid identifier".format(k=k))
         if keyword.iskeyword(k):
             raise ValueError("{k} is a python keyword".format(k=k))
```
Comment (jbrockmendel) [inline pandas/_config/config.py]: use fstrings while in the neighborhood?

**Comment by jbrockmendel** [comment]
small comments, non-blockers, LGTM

**Comment by jbrockmendel** [comment]
LGTM

File: pandas/_config/config.py [hunk 3]
```diff
@@ -455,10 +453,12 @@ def register_option(key, defval, doc="", validator=None, cb=None):
     path = key.split(".")
 
     for k in path:
-        if not bool(re.match("^" + tokenize.Name + "$", k)):
-            raise ValueError("{k} is not a valid identifier".format(k=k))
+        # NOTE: tokenize.Name is not a public constant
+        # error: Module has no attribute "Name"  [attr-defined]
+        if not re.match("^" + tokenize.Name + "$", k):  # type: ignore
```
Comment (WillAyd) [inline pandas/_config/config.py]: Maybe worth considering `tokenize.Name` to `\w+`; it hasn't changed in 9 years:

https://github.com/python/cpython/blob/305189ecdc8322c22879a04564cad5989f937462/Lib/tokenize.py#L67

**Review by WillAyd — APPROVED** [review]
lgtm. optional comment to maybe address or merge if you disagree

**Comment by jreback** [comment]
thanks @simonjayhawkins comment might be worth a followup

**Comment by simonjayhawkins** [comment]
> thanks @simonjayhawkins, @WillAyd comment might be worth a followup

The ignore was removed in #36012, since Name was added to the stub for tokenize.

I don't think a follow-up now needed or we need to change our usage.

**Pull request merged**